app/lib/features/tasks/widgets/task_list_card.dart

```diff
@@ -0,0 +1,239 @@
+import 'package:acter/features/home/widgets/space_chip.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:acter/common/snackbars/custom_msg.dart';
+import 'package:acter/features/tasks/widgets/task_entry.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import 'package:flutter/material.dart';
+
+@immutable
+class TasksOverview {
+  final List<Task> openTasks;
+  final List<Task> doneTasks;
+  const TasksOverview({required this.openTasks, required this.doneTasks});
+}
+
+class TasksNotifier extends FamilyAsyncNotifier<TasksOverview, TaskList> {
+  late Stream<void> subscriber;
+
+  Future<TasksOverview> _refresh(TaskList taskList) async {
+    final tasks = (await taskList.tasks()).toList();
+    List<Task> openTasks = [];
+    List<Task> doneTasks = [];
+    for (final task in tasks) {
+      if (task.isDone()) {
+        doneTasks.add(task);
+      } else {
+        openTasks.add(task);
+      }
+    }
+
+    // FIXME: ordering?
+
+    return TasksOverview(openTasks: openTasks, doneTasks: doneTasks);
+  }
+
+  @override
+  Future<TasksOverview> build(TaskList arg) async {
+    // Load initial todo list from the remote repository
+    final taskList = arg;
+    final retState = _refresh(taskList);
+    subscriber = taskList.subscribeStream();
+    subscriber.forEach((element) async {
+      state = const AsyncValue.loading();
+      state = await AsyncValue.guard(() async {
+        return await _refresh(taskList);
+      });
+    });
+    return retState;
+  }
+}
+
+final tasksProvider =
+    AsyncNotifierProvider.family<TasksNotifier, TasksOverview, TaskList>(() {
+  return TasksNotifier();
+});
+
+class InlineTaskAdd extends StatefulWidget {
```

Comment:
What order should it have? And I assume by private you mean starting with an underscore?

Reply:
By ordering meant the widget should be at the top on the basis of the file naming .i.e. task_list_card. Providers should be private if only used here since they don't need to be kept global if not shared state. Otherwise too much common naming might conflict with other providers, unintentionally..